FIXES

omar-limbert opened the pull request:
- C.R.U.D. on lightning theme is working, delete coming soon.
- Some selector fixes.
- Time fixed to stability.

Commit:
FIXES

- C.R.U.D. on lightning theme is working, delete coming soon.
- Some selector fixes.
- Time fixed to stability.

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -39,17 +40,25 @@ private CommonActions() {
      */
     public static WebElement getElement(final WebElement element) {
-        WAITER.until(ExpectedConditions.visibilityOf(element));
+        WAITER.until(ExpectedConditions.elementToBeClickable(element));
```
Are you sure to do this ?
Review how this will affect in the framework.

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -183,17 +190,19 @@ public static String getConfirmMessageShowed(final WebElement element) {
                 .stream()
                 .filter(elementOnList -> elementOnList.getText().equalsIgnoreCase(element))
                 .findFirst()
                 .orElse(null);
```
It is not a good practice to return null values

Commit:
Fixes

C.R.U.D. on light theme is working, but need implement to 2nd button for edit and delete.

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -177,23 +185,39 @@ public static void autoCompleterLightTheme(final String textToSelect) {
+    public static WebElement getWebElementFromAList(final List<WebElement> listOfElements, final String element) {
+        waitTime(2);
+
+        WebElement webElement = listOfElements
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

carledriss commented on src/main/java/org/fundacionjala/sfdc/pageobjects/SFMain.java:
```diff
@@ -50,9 +51,9 @@ public void clickElementOnList(final String elementOnList) {
      * @return getAccountHomePage.
      */
-    public WebElement getElementOnList(final String elementOnList) {
+    public boolean getElementOnList(final String elementOnList) {
```
Please rename the method name

Commit:
fix

fix

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -39,17 +40,26 @@ private CommonActions() {
+     * @param time .
+     */
+    public static void waitTime(int time) {
+        WEB_DRIVER.manage().timeouts().implicitlyWait(time, TimeUnit.SECONDS);
```
@AT-06/at-06 
This change will have problems in the future.
if you are changing the implicit wait or explicit wait 
you should ensure to restore by the default value.

Consider using a try - finally

Commit:
fixed

fixed

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -209,6 +242,4 @@ public static void closeMessageLighting() {
             System.out.println("Message not displayed.");
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

carledriss commented:
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
fix

fix

Commit:
properties

properties

carledriss commented on src/main/java/org/fundacionjala/sfdc/util/CommonActions.java:
```diff
@@ -205,10 +239,6 @@ public static void closeMessageLighting() {
-        } catch (NoSuchElementException e) {
-            System.out.println("Message not displayed.");
-        }
+        } catch (NoSuchElementException e) { }
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00108)

Commit:
fix

fix

omar-limbert committed:
Merge branch 'develop' into FIXES

omar-limbert committed:
fix

fix
